### PR TITLE
notebooks for evaluating CESM ensemble spread and additional libraries to the analysis env

### PR DESCRIPTION
**Pull request by jhayron-perez**
I added the notebook for evaluating CESM ensemble spread and the .py with the necessary functions. I also added scipy, scikit-learn, properscoring to the .yml file. This is just an initial version, I still have to do more tests related, for example, to the number of bins used in some of the metrics.

### CODE REVIEW

File: environment-analysis.yml [hunk 1]
```diff
@@ -16,3 +16,6 @@ dependencies:
   - xarray=2023.3.0
   - xesmf=0.7.1
   - xskillscore=0.0.24
+  - scipy=1.9.1
```
Comment (katiedagon) [inline environment-analysis.yml]: Thanks for adding these packages! I suggest keeping the list in alphabetical order to match the other yml files.

Comment (jhayron-perez) [reply environment-analysis.yml]: Great! I'll change it.

**Comment by jhayron-perez** [comment]
<p>Oh I actually hadn't noticed this error since I was focusing on reviewing the temperature notebook. It seems that the files in one folder are incomplete. I'll tweak the code so it skips the member (and warns the user) when this happens and check if we can complete the number of files in this specific folder :)</p>

---

 <a href='https://app.reviewnb.com/mariajmolina/s2sml/pull/14/discussion/#comment-1681274146'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by katiedagon** [comment]
<p>Sounds good. Do you think this incomplete member folder is also impacting the ensemble means used to train the ML models?</p>

---

 <a href='https://app.reviewnb.com/mariajmolina/s2sml/pull/14/discussion/#comment-1681274146'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by jhayron-perez** [comment]
<p>Good point! I think the code that computes these files is a modification of the code that computes the mean, but this is Maria's field. I'll let her know and ask about it, it may be possible that for the missing dates the mean is only being computed with one less member.</p>

---

 <a href='https://app.reviewnb.com/mariajmolina/s2sml/pull/14/discussion/#comment-1681274146'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by mariajmolina** [comment]
<p>Thank you for catching this! Yes -- looks like one of the ensemble members may have not finished processing -- we now have 1195 files for the m10 member</p>

---

 <a href='https://app.reviewnb.com/mariajmolina/s2sml/pull/14/discussion/#comment-1681274146'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by mariajmolina** [comment]
<p>Regarding the question about ensemble means -- the ensemble means were calculated without using these individual files -- these were calculated after the fact for the ensemble calculation -- so we should be fine with that</p>

---

 <a href='https://app.reviewnb.com/mariajmolina/s2sml/pull/14/discussion/#comment-1681274146'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

File: environment-analysis.yml (hunk 1 above)
Comment (mariajmolina) [reply environment-analysis.yml]: @jhayron-perez go ahead and push that change using git -- it should update this pull request with your change

File: notebooks/EnsembleEvaluation/EvaluateEnsemble_CESM_Precipitation.ipynb
Comment (kjmayer) [inline notebooks/EnsembleEvaluation/EvaluateEnsemble_CESM_Precipitation.ipynb:1]: The 'jet' colormap generally ins't colorblind friendly... I don't think its a necessity to change in this group, but it should be changed at some point before publication.

**Review by mariajmolina — APPROVED** [review]

**Pull request merged**